public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/NewDimensionFields.tsx

```diff
@@ -0,0 +1,207 @@
+          labelOptions.push({ value: filter, label: filter });
+        }
+      }
+    }
+    return labelOptions;
+  };
+  const getValidOperators = (selectedOperator: string) => {
+    if (dimensionOperators.find((operator: SelectableValue) => operator.value === selectedOperator)) {
+      return dimensionOperators;
+    }
+    return [...dimensionOperators, ...(selectedOperator ? [{ label: selectedOperator, value: selectedOperator }] : [])];
+  };
+
+  const changedFunc = (changed: Array<Partial<AzureMetricDimension>>) => {
+    const properData: AzureMetricDimension[] = changed.map((x) => {
+      return {
+        dimension: x.dimension ?? '',
+        operator: x.operator ?? 'eq',
+        filters: x.filters ?? [],
+      };
+    });
+    onQueryChange(setDimensionFilters(query, properData));
+  };
+
+  const renderFilters = (
+    item: Partial<AzureMetricDimension>,
+    onChange: (item: Partial<AzureMetricDimension>) => void,
+    onDelete: () => void
+  ) => {
+    return (
+      <HorizontalGroup spacing="none">
+        <Select
+          menuShouldPortal
+          placeholder="Field"
+          value={item.dimension}
+          options={getValidDimensionOptions(item.dimension || '')}
+          onChange={(e) => onFieldChange('dimension', item, e.value ?? '', onChange)}
+        />
+        <Select
+          menuShouldPortal
+          placeholder="Operation"
+          value={item.operator}
+          options={getValidOperators(item.operator || 'eq')}
+          onChange={(e) => onFieldChange('operator', item, e.value ?? '', onChange)}
+          allowCustomValue
+        />
+        {item.operator === 'eq' || item.operator === 'ne' ? (
+          <MultiSelect
+            menuShouldPortal
+            placeholder="Select value(s)"
+            value={item.filters}
+            options={getValidMultiSelectOptions(item.filters, item.dimension ?? '')}
+            onChange={(e) =>
+              onFieldChange(
+                'filters',
+                item,
+                e.map((x) => x.value ?? ''),
+                onChange
+              )
+            }
+            aria-label={'dimension-labels-select'}
+            allowCustomValue
+          />
+        ) : (
+          // The API does not currently allow for multiple "starts with" clauses to be used.
+          <Select
+            menuShouldPortal
+            placeholder="Select value"
+            value={item.filters ? item.filters[0] : ''}
+            allowCustomValue
+            options={getValidFilterOptions(item.filters ? item.filters[0] : '', item.dimension ?? '')}
+            onChange={(e) => onFieldChange('filters', item, [e?.value ?? ''], onChange)}
+            isClearable
+          />
+        )}
+        <AccessoryButton aria-label="Remove" icon="times" variant="secondary" onClick={onDelete} type="button" />
+      </HorizontalGroup>
+    );
+  };
+
+  return (
+    <Field label="Dimensions">
```

Comment:
Should this be imported from `public/app/plugins/datasource/grafana-azure-monitor-datasource/components/Field.tsx` instead of `@grafana/ui`.

Reply:
Good catch!